cosmos_transfer1/diffusion/datasets/example_transfer_dataset.py

```
@@ -226,15 +269,210 @@ def __str__(self):
+        assert (np.array(frame_ids) < len(vr)).all()
+        assert (np.array(frame_ids) >= 0).all()
+        vr.seek(0)
+        frame_data = vr.get_batch(frame_ids).asnumpy()
+        try:
+            fps = vr.get_avg_fps()
+        except Exception:  # failed to read FPS
+            fps = 24
+        return frame_data, fps
+
+
+    def __getitem__(self, index):
+        max_retries = 3
+        for _ in range(max_retries):
+            try:
+                video_path = self.video_paths[index]
+                video_name = os.path.basename(video_path).replace(".mp4", "")
+
+                data = dict()
+                ctrl_videos = []
+                videos = []
+                t5_embeddings = []
+                view_indices = [i for i in range(len(self.view_keys))]
+                view_indices_conditioning = []
+                if self.sample_n_views > 1:
+                    sampled_idx = np.random.choice(
+                        np.arange(1, len(view_indices)), size=min(self.sample_n_views - 1, len(view_indices) - 1),
+                        replace=False
+                    )
+                    sampled_idx = np.concatenate(
+                        [
+                            [
+                                0,
+                            ],
+                            sampled_idx,
+                        ]
+                    )
+                    sampled_idx.sort()
+                    view_indices = sampled_idx.tolist()
+
+                frame_ids = None
+                fps = None
+                for view_index in view_indices:
+                    view_key = self.view_keys[view_index]
+                    if frame_ids is None:
+                        frames, frame_ids, fps = self._sample_frames(video_path)
+                        if frames is None:  # Invalid video or too short
+                            index = np.random.randint(len(self.video_paths))
+                            continue
```

We should raise an exception here. This continue statement is inside a view_index loop nested under the outer max_retries loop so it will have the impact of loading the next view from the same video_path, instead of moving on to the next video_path at the newly sampled index.
The catch block will reset `index = np.random.randint(len(self.video_paths))` as well so we can also remove it from here.